stl/inc/ranges

```diff
@@ -3589,9 +3608,214 @@ namespace ranges {
     };
 
     template <class _Rng, class _Pat>
-    split_view(_Rng&&, _Pat&&) -> split_view<views::all_t<_Rng>, views::all_t<_Pat>>;
+    lazy_split_view(_Rng&&, _Pat&&) -> lazy_split_view<views::all_t<_Rng>, views::all_t<_Pat>>;
 
     template <input_range _Rng>
+    lazy_split_view(_Rng&&, range_value_t<_Rng>)
+        -> lazy_split_view<views::all_t<_Rng>, single_view<range_value_t<_Rng>>>;
+
+    namespace views {
+        // VARIABLE views::lazy_split
+        class _Lazy_split_fn {
+        private:
+            template <class _Delim>
+            struct _Partial : _Pipe::_Base<_Partial<_Delim>> {
+                /* [[no_unique_address]] */ _Delim _Delimiter;
+
+                template <viewable_range _Rng>
+                _NODISCARD constexpr auto operator()(_Rng&& _Range) const& noexcept(
+                    noexcept(lazy_split_view{_STD forward<_Rng>(_Range), _Delimiter})) requires requires {
+                    lazy_split_view{static_cast<_Rng&&>(_Range), _Delimiter};
+                }
+                { return lazy_split_view{_STD forward<_Rng>(_Range), _Delimiter}; }
+

```

Comment:
This will need a `const&& = delete;` overload to approximate perfect forwarding call wrapper behavior.

Reply:
`_Delim` is a program-defined type; how can we conformingly fail to implement all four overloads of `operator()`? (Ditto for `lazy_split`'s range adaptor closure.)

Reply:
Conformingly, you'll need 8. See https://reviews.llvm.org/D107098#2915589 and https://reviews.llvm.org/D107199.

Reply:
I've added an [action item](https://github.com/microsoft/STL/projects/9#card-62754027) to track fixing this in a followup PR.